### DISCUSSION TITLE
Resilience4J + TimeLimiter + Bulkhead

### DISCUSSION THREAD

**Discussion by keshoresiva** [resilience4j/resilience4j#2159] [General]
Hello.. I am trying to implement R4j with timelimiter. When my main method takes more time.. fallback is getting executed. In this approach, is the future thread getting cancelled as it took more time ..?  I am using java 11. Please advice.

i am having below property 
resilice4j.timelimiter.configs.tl.cancel-running-future=true
resilice4j.timelimiter.configs.tl.timeout-duration=10ms

@Bulkhead(name="bh", type=Bulkhead.Type.THREAPOOL)
@TimeLimiter(name="tl", fallbackMethod="fixissue")
@CircuitBraker(name="cb", fallbackMethod="fixissue")
public CompletableFuture<List<Product>> getItem(){
             // biz logic and calls to db
}

public CompletableFuture<Llist<Product> fixissue(Throwable t) {
           
}

**Comment by RobWin** [comment]
CompletableFuture cannot be canceled.

**Reply by achoice** [reply]
Can't it? There is a cancel method on it;
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/CompletableFuture.html#cancel(boolean)
